src/store/reducers/SavedQueries.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  QueryType,
+  SavedQueriesActions,
+  SavedQueriesActionTypes,
+  SavedQueriesState,
+  SavedQueriesStatuses,
+} from '../SavedQueriesTypes';
+
+const initialState: SavedQueriesState = {
+  isLoadingAllQueries: false,
+  queries: { [QueryType.file]: [], [QueryType.cohort]: [] },
+  queryIdToStatus: {},
+  errorFetchAllQueries: null,
+};
+
+export default (state = initialState, action: SavedQueriesActionTypes): SavedQueriesState => {
+  switch (action.type) {
+    case SavedQueriesActions.toggleFetchAllSavedQueries: {
+      return { ...state, isLoadingAllQueries: action.isLoadingAllQueries };
+    }
+    case SavedQueriesActions.addAllSavedQueries: {
+      return { ...state, queries: action.queries };
+    }
+    case SavedQueriesActions.failureFetchingAllSavedQueries: {
+      return { ...state, errorFetchAllQueries: action.error };
+    }
+    case SavedQueriesActions.requestDeleteSavedQuery: {
+      const queryId = action.queryId;
+
+      const queryIdToStatusShallowCopy = { ...state.queryIdToStatus };
+      queryIdToStatusShallowCopy[queryId] = SavedQueriesStatuses.deleting;
+
+      return { ...state, queryIdToStatus: queryIdToStatusShallowCopy };
+    }
+    case SavedQueriesActions.successDeletingSavedQuery: {
+      const queryId = action.queryId;
```

Comment:
I usually avoid to have complex logic in reducer cases, I would probably add some utils function to remove elem from a list/ a map (or if we have kind libs like lodash, i would use them), and just call them here (moreover, they could be useful in others places of the app)

Reply:
Good point. I'll change that.